### PR TITLE
Test application factory preserves environment name set by variable.

### PR DESCRIPTION
**Pull request by henrik-olsson-valtech**
- If it has not been set, use "Development" as default as before.
- Also, add tests for these two cases.

Addresses #7976

### CODE REVIEW

**Comment by natemcmaster** [comment]
My automation detected this was targeting the dev branch and updated the PR to target master. We'll be deleting dev branches soon so please use master instead. Thanks!

**Comment by mkArtakMSFT** [comment]
@javiercn, @dougbu can you please review this? Thanks!

File: test/Microsoft.AspNetCore.Mvc.FunctionalTests/TestingInfrastructureEnvironmentTests.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        [Fact]
+        public void TestingInfrastructure_PreservesEnvironmentSetByVariable()
+        {
+            var factory = new StagingEnvironmentWebApplicationFactory();
+            var client = factory.CreateClient();
+            Assert.Equal(EnvironmentName.Staging, factory.EnvironmentName);
+        }
+    }
+
+    public class DefaultEnvironmentWebApplicationFactory : WebApplicationFactory<BasicWebSite.Startup>
```
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/TestingInfrastructureEnvironmentTests.cs]: Make these classes `private`

Comment (dougbu) [reply test/Microsoft.AspNetCore.Mvc.FunctionalTests/TestingInfrastructureEnvironmentTests.cs]: Oh, and move them inside `TestingInfrastructureEnvironmentTests`

**Comment by dougbu** [comment]
@henrik-olsson-valtech I changed the base branch of this PR to target `release/2.2`. However your PR now contains extra commits. Please rebase on `origin/release/2.2` and `git push --force` to clear that up.

**Comment by javiercn** [comment]
Hi,

as commented on https://github.com/aspnet/Mvc/issues/7976#issuecomment-403577967 this is by design. To enable this scenario you should override ConfigureWebHost to setup the appropriate environment for your tests.

File: test/Microsoft.AspNetCore.Mvc.FunctionalTests/TestingInfrastructureEnvironmentTests.cs (hunk 1 above)
Comment (henrik-olsson-valtech) [reply test/Microsoft.AspNetCore.Mvc.FunctionalTests/TestingInfrastructureEnvironmentTests.cs]: OK, will do.

**Comment by henrik-olsson-valtech** [comment]
@dougbu I used the dev branch because [How to contribute](https://github.com/aspnet/Home/blob/master/CONTRIBUTING.md) says "Pull requests should all be done to the dev branch (not master).". Should that be changed, then?

**Comment by henrik-olsson-valtech** [comment]
@javiercn OK, if it is by design, then I would suggest that goes into the documentation. I read both [Integration tests in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/test/integration-tests?view=aspnetcore-2.1#basic-tests-with-the-default-webapplicationfactory) and [WebApplicationFactory<TEntryPoint> Class](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.testing.webapplicationfactory-1?view=aspnetcore-2.1) and did not find anything.

**Comment by dougbu** [comment]
> I used the dev branch because [How to contribute](https://github.com/aspnet/Home/blob/master/CONTRIBUTING.md) says &hellip; Should that be changed, then?

Yes. Thanks for pointing that out.

**Pull request closed without merging**